eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -396,7 +397,8 @@ object Router {
                           ignore: Ignore = Ignore.empty,
                           routeParams: Option[RouteParams] = None,
                           allowMultiPart: Boolean = false,
-                          pendingPayments: Seq[Route] = Nil)
+                          pendingPayments: Seq[Route] = Nil,
+                          paymentConfig: Option[SendPaymentConfig] = None)
```

Comment:
It's a bit overkill to send the whole `SendPaymentConfig` whereas what we need is just the ids and the `payment_hash`...
We may in the future wrongly start using other fields of this class, I'd prefer sending only what we need but maybe I'm nit-picking a bit too much :)

Reply:
How about with a comment like in b1c32b7a6?

Reply:
I'd prefer something like `paymentContext: Option[PaymentContext]` where `case class PaymentContext(id: UUID, parentId: UUID, paymentHash: ByteVector32)` that's easily built from the `SendPaymentConfig`